### DISCUSSION TITLE
How to use createCommitOnBranch?

### DISCUSSION THREAD

**Discussion by frmoded** [community/community#24599] [Apps, API and Webhooks]
<p>I am trying to create a new commit using the github graphql api, using the <a href="https://github.blog/changelog/2021-09-13-a-simpler-api-for-authoring-commits/" rel="noopener nofollow ugc">createCommitOnBranch</a> mutation. What value should one use for expectedHeadOid: “???”? How can one get such value from the Graphql API?</p>
<p>This is my attempt so far:</p>
<p>{ mutation m1 { createCommitOnBranch( input: { branch: {repositoryNameWithOwner: “some_repo/some_owner”, branchName: “main” }, message: {headline: “headline!”}, fileChanges: { additions: {path: “README.md”, contents: “SGVsbG8gV29ybGQ=”} } expectedHeadOid: “???” } ) } }</p>

**Comment by frmoded** [comment, marked as answer]
<p>Resolved, see:</p><aside class="onebox stackexchange" data-onebox-src="https://stackoverflow.com/questions/72836597/how-to-create-new-commit-with-the-github-graphql-api">
  <header class="source">

      <a href="https://stackoverflow.com/questions/72836597/how-to-create-new-commit-with-the-github-graphql-api" target="_blank" rel="noopener nofollow ugc">stackoverflow.com</a>
  </header>

  <article class="onebox-body">
      <a href="https://stackoverflow.com/users/11639182/oded" target="_blank" rel="noopener nofollow ugc">
    <img alt="Oded" src="https://user-images.githubusercontent.com/6885133/181085564-cf58d5cf-e718-4125-9937-82e54b874f7c.png" class="thumbnail onebox-avatar" width="256" height="256">
  </a>

<h4>
  <a href="https://stackoverflow.com/questions/72836597/how-to-create-new-commit-with-the-github-graphql-api" target="_blank" rel="noopener nofollow ugc">How to create new commit with the github graphql API?</a>
</h4>

<div class="tags">
  <strong>github, graphql</strong>
</div>

<div class="date">
  asked by
  
  <a href="https://stackoverflow.com/users/11639182/oded" target="_blank" rel="noopener nofollow ugc">
    Oded
  </a>
  on <a href="https://stackoverflow.com/questions/72836597/how-to-create-new-commit-with-the-github-graphql-api" target="_blank" rel="noopener nofollow ugc">05:05AM - 02 Jul 22 UTC</a>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by tuves** [comment]
<p>Thank you for circling back here with the solution, <a class="mention" href="/frmoded">@frmoded</a>!</p>
